src/build/java/build/please/compile/FileFinder.java

```diff
@@ -0,0 +1,48 @@
+package build.please.compile;
+
+import java.nio.file.FileVisitResult;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.nio.file.FileVisitResult.CONTINUE;
+
+class FileFinder extends SimpleFileVisitor<Path> {
+
+  private final String extension;
+  private final List<String> files = new ArrayList<>();
```

Comment:
Not sure what people do with this list once it is returned by getFiles, but technically a LinkedList would save you any internal array resizing costs if adding many files here. (Practically though it is probably not worth it).